### ISSUE TITLE
[java] NonSerializableClass FP with enum

### ISSUE THREAD

**Issue by Carlson-JLQ** [pmd/pmd#6534]
# False Positive: Acceptable fields and benchmark exception cases are still treated as non-serializable-class problems.

Version
7.22.0

## Checker
- Checker id: `category/java/errorprone.xml/NonSerializableClass`
- Checker description: Detects non-serializable fields in Serializable classes that are not handled by custom serialization methods or serialPersistentFields.

## Description of the false positive
`diff/output` marks 1 false positives for `category/java/errorprone.xml/NonSerializableClass`. Across these samples, the checker tends to fire on the outer syntax while missing code-level cues that make the pattern harmless, exempted, or at least much weaker than a real bug. The reports read more like over-eager matching than strong bug findings.

### `NegCase8.java`
Source: `output/codeql/NonSerializableField/src/main/java/scensct/core/neg/NegCase8.java`

The sample keeps the benchmark's target pattern but expresses it in a slightly non-canonical source form. A human reader would likely classify this as an allowed or intentionally structured pattern rather than a bug. That matches the benchmark's stated intent: A serializable class (via inheritance) that is an enum type, containing a non-transient, non-static field of a non-serializable type should not be flagged as violation.

```java
// A serializable class (via inheritance) that is an enum type, containing a non-transient, non-static field of a non-serializable type should not be flagged as violation.
package scensct.core.neg;

import java.io.Serializable;

public class NegCase8 {
    // Non-serializable type
    private static class NonSerializableType {}
    
    // Serializable enum type
    public enum MyEnum implements Serializable {
        VALUE1, VALUE2;
        
        // Non-transient, non-static field of non-serializable type in enum
        private NonSerializableType field;
    }
}
```

## Cause analysis
The rule is documented as follows: Detects non-serializable fields in Serializable classes that are not handled by custom serialization methods or serialPersistentFields.

 The implementation appears to lean too heavily on surface indicators and not enough on what the code actually does.

That is why these reports read like over-approximation. The code often retains a suspicious signature or control-flow skeleton, but once you read the body, the benchmark's claimed exemption is visible in the source itself.

For `category/java/errorprone.xml/NonSerializableClass`, the practical improvement would be to model those source-level exemptions more explicitly instead of reporting every syntactic near-match.

## References
None known.

**Comment by zbynek** [comment]
@Carlson-JLQ 
> A serializable class (via inheritance) that is an enum type, containing a non-transient, non-static field of a non-serializable type should not be flagged as violation.

Wrong.

I f you think there's an issue with the `NonSerializableClass` rule, please describe it with your own words (without AI).
